### PR TITLE
make from_bytes is decorated by data_struct

### PR DESCRIPTION
**Pull request by stevenlaw123**
make from_bytes is decorated by data_struct and provides the bcs_friend module, addressing the issue of not being able to use generic parameters in moveos_std::any.

### CODE REVIEW

File: crates/rooch-framework/sources/bcs_friend.move [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// Source from https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/from_bcs.move
+
+/// This module provides a number of functions to convert _primitive_ types from their representation in `std::bcs`
+/// to values. This is the opposite of `bcs::to_bytes`. Note that it is not safe to define a generic public `from_bytes`
+/// function because this can violate implicit struct invariants, therefore only primitive types are offerred. If
+/// a general conversion back-and-force is needed, consider the `moveos_std::Any` type which preserves invariants.
+module rooch_framework::bcs_friend {
```
Comment (jolestar) [inline crates/rooch-framework/sources/bcs_friend.move:10]: We do not need to add a new module for this, we just need a function. And the rooch_framework do not need a bcs module.

File: crates/rooch-framework/sources/ethereum_light_client.move [hunk 2]
```diff
@@ -58,7 +58,7 @@ module rooch_framework::ethereum_light_client{
 
     fun process_block(ctx: &mut Context, block_header_bytes: vector<u8>){
         //TODO find a way to deserialize the block header, do not use bcs::from_bytes
-        let block_header = bcs::from_bytes<BlockHeader>(block_header_bytes);
+        let block_header = bcs_friend::from_bytes<BlockHeader>(block_header_bytes);
```
Comment (jolestar) [inline crates/rooch-framework/sources/ethereum_light_client.move:61]: Make BlockHeader to `data_struct`

File: moveos/moveos-stdlib/moveos-stdlib/sources/bcs_friend.move [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// Source from https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/from_bcs.move
+
+/// This module provides a number of functions to convert _primitive_ types from their representation in `std::bcs`
+/// to values. This is the opposite of `bcs::to_bytes`. Note that it is not safe to define a generic public `from_bytes`
+/// function because this can violate implicit struct invariants, therefore only primitive types are offerred. If
+/// a general conversion back-and-force is needed, consider the `moveos_std::Any` type which preserves invariants.
+module moveos_std::bcs_friend{
```
Comment (jolestar) [inline moveos/moveos-stdlib/moveos-stdlib/sources/bcs_friend.move:10]: Same as previous

**Review by jolestar — CHANGES_REQUESTED** [review]

File: crates/rooch-framework/sources/bcs_friend.move (hunk 1 above)
Comment (stevenlaw123) [reply crates/rooch-framework/sources/bcs_friend.move:10]: This PR has a lot of files, I have already addressed these issues in another PR.

**Pull request closed without merging**